### PR TITLE
chore: add tests for app theming

### PR DESCRIPTION
**Pull request by jsartisan**
/ok-to-test tags="@tag.Anvil"

## Summary by CodeRabbit

- **New Features**
  - Introduced theme setting and snapshot verification for various widgets in different modes (Canvas, Preview, Deploy).
  - Added a new CSS rule to enhance the styling of input elements based on specific conditions.

- **Refactor**
  - Renamed existing snapshot verification methods to more descriptive names across multiple widget snapshot files.
  - Added methods to handle theme settings and detailed snapshot matching.

- **Tests**
  - Enhanced testing functionality for widgets with updated methods for snapshot verification across different themes.

These updates ensure more detailed and theme-specific snapshot testing for various widget states, improving test coverage and accuracy, as well as enhancing the layout consistency of input elements.

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/10046186626>
> Commit: a6232ae795f77bd46bff4c854f46c5671a518ff6
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=10046186626&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Anvil`
> Spec:
> <hr>Mon, 22 Jul 2024 18:43:52 UTC

### CODE REVIEW

File: app/client/cypress/support/Pages/Anvil/AnvilSnapshot.ts [hunk 1]
```diff
@@ -1,91 +1,138 @@
...
+
   private locators = {
     enterPreviewMode: ObjectsRegistry.CommonLocators._enterPreviewMode,
     exitPreviewMode: ObjectsRegistry.CommonLocators._exitPreviewMode,
     canvas: "[data-testid=t--canvas-artboard]",
     colorMode: "[data-testid=t--anvil-theme-settings-color-mode]",
     appViewerPage: "[data-testid=t--app-viewer-page]",
     propertyPaneSidebar: "[data-testid=t--property-pane-sidebar]",
+    accentColorInput: "[data-testid=t--color-picker-input]",
+    fontFamilySectionInput: "[data-testid=t--anvil-theme-settings-typography] input",
```
Comment (KelvinOm) [inline app/client/cypress/support/Pages/Anvil/AnvilSnapshot.ts:17]: Can we set the `data-testid` directly on needed components. When using locators, we should avoid class names or tags.

**Review by KelvinOm — CHANGES_REQUESTED** [review]

**Comment by sagar-qa007** [comment]
Please all test to verify changes.

**Comment by jsartisan** [comment]
@KelvinOm Let me know if we can merge this PR.

**Review by KelvinOm — APPROVED** [review]

**Pull request merged**